Scripts/level/sense/patterns/customWallStuff.lua

```diff
@@ -0,0 +1,159 @@
+	cwHandle = cw_create()
+	cw_setVertexColor(cwHandle, 0, mRed, mGreen, mBlue, mAlpha)
+	cw_setVertexColor(cwHandle, 1, mRed, mGreen, mBlue, mAlpha)
+	cw_setVertexColor(cwHandle, 2, mRed, mGreen, mBlue, mAlpha)
+	cw_setVertexColor(cwHandle, 3, mRed, mGreen, mBlue, mAlpha)
+	cw_setVertexPos(cwHandle, 0, 0, 0)
+	cw_setVertexPos(cwHandle, 1, 0, 0)
+	cw_setVertexPos(cwHandle, 2, 0, 0)
+	cw_setVertexPos(cwHandle, 3, 0, 0)
+	
+	
+	
+	iw = imposterWall:new(cwHandle)
+	-- set up The Shit™
+	iw.distance = mDistance
+	iw.speed = mSpeed
+	iw.side = mSide
+	iw.shape = mShape
+	iw.thickness = mThickness
+	table.insert(imposterWalls, iw)
+end
+
+-- common stuff like cImpWall and patterns in imposterpatterns.lua
+
+-- ACTUAL LEVEL BELOW
+
+-- this function adds a pattern to the timeline based on a key
+function addPattern(mKey)
+		if mKey == 0 then pImpAltBarrage(0, 2)
+	elseif mKey == 1 then pImpRandomBarrage(0, 2.5) 
+	elseif mKey == 2 then
+	elseif mKey == 3 then 
+	end
+end
+
+-- shuffle the keys, and then call them to add all the patterns
+-- shuffling is better than randomizing - it guarantees all the patterns will be called
+keys = {  }
+keys = shuffle(keys)
+index = 0
+
+-- onInit is an hardcoded function that is called when the level is first loaded
+function onInit()
+	l_setSpeedMult(1)
+	l_setSpeedInc(0.045)
+	l_setSpeedMax(2.35);
+	l_setRotationSpeed(0.17)
+	l_setRotationSpeedMax(0.45)
+	l_setRotationSpeedInc(0.045)
+	l_setDelayMult(1.8)
+	l_setDelayInc(-0.01)
+	l_setDelayMin(1.07)
+	l_setFastSpin(71.0)
+	l_setSides(6)
+	l_setSidesMin(6)
+	l_setSidesMax(6)
+	l_setIncTime(15)
+	
+	-- debug! only use for testing wall spawns
+	--l_setPulseMin(20)
+	--l_setRadiusMin(20)
+	
+	l_setIncEnabled(false)
+end
+
+
+
+-- onLoad is an hardcoded function that is called when the level is started/restarted
+function onLoad()
+	l_setRotationSpeed(l_getRotationSpeed() * getRandomDir())
+	--cssImpWall(0)
+	pImpRandomBarrage(0, 2.5) 
+end
+
+-- onStep is an hardcoded function that is called when the level timeline is empty
+-- onStep should contain your pattern spawning logic
+function onStep()
+	addPattern(keys[index])
+	index = index + 1
+
+	if index - 1 == #keys then
+		index = 1
+		keys = shuffle(keys)
+	end
+end
+
+-- onIncrement is an hardcoded function that is called when the level difficulty is incremented
+function onIncrement()
+end
+
+-- onUnload is an hardcoded function that is called when the level is closed/restarted
+function onUnload()
+end
+
+-- onUpdate is an hardcoded function that is called every frame
+function onUpdate(mFrameTime)
+	-- remove dead walls (walls whose distance is 0)
+	ArrayRemove(imposterWalls, function(t, i, j)
+		local v = t[i]
+		if v.dead then
+			cw_destroy(v.cwHandle)
+			return true
+		else
+			return false
+		end
+	end);
+	
+	-- Loop through all imposter walls to handle their movement
+	for _, iw in ipairs(imposterWalls) do
+		iw:move(mFrameTime)
+	end
+end

```

Comment:
Duplicates

Reply:
remove the `customWallStuff.lua` please, it's redundant @zxdasfsa